src/main/java/com/uid2/admin/vertx/service/ServiceLinkService.java

```diff
@@ -107,11 +109,33 @@ private void handleServiceLinkAdd(RoutingContext rc) {
                 return;
             }
 
+            if (name.isEmpty()) {
+                ResponseUtil.error(rc, 400, "name cannot be empty");
+                return;
+            }
+
+            Set<Role> serviceRoles = serviceProvider.getService(serviceId).getRoles();
+            final Set<Role> roles;
+            if (rolesJson == null || rolesJson.isEmpty()) {
```

Comment:
We can reduce nesting in this block

Reply:
You may also consider extracting this into a function.